risc0/build/src/docker.rs

```diff
@@ -76,21 +95,62 @@ pub fn docker_build(manifest_path: &Path, src_dir: &Path, features: &[String]) -
         let temp_path = temp_dir.path();
         let rel_manifest_path = manifest_path.strip_prefix(&src_dir)?;
         create_dockerfile(rel_manifest_path, temp_path, pkg_name.as_str(), features)?;
-        build(&src_dir, temp_path)?;
+        build(src_dir.as_path(), temp_path)?;
     }
     println!("ELFs ready at:");
 
-    let target_dir = src_dir.join(TARGET_DIR);
-    for target in root_pkg.targets.iter() {
+    for target in get_targets(&root_pkg) {
         if target.is_bin() {
-            let elf_path = target_dir.join(&pkg_name).join(&target.name);
+            let elf_path = get_elf_path(&src_dir, &pkg_name, &target.name);
             let image_id = compute_image_id(&elf_path)?;
             let rel_elf_path = Path::new(TARGET_DIR).join(&pkg_name).join(&target.name);
             println!("ImageID: {} - {:?}", image_id, rel_elf_path);
         }
     }
 
-    Ok(())
+    Ok(BuildStatus::Success)
+}
+
+fn canonicalize_path(path: &Path) -> Result<PathBuf> {
+    path.canonicalize()
+        .context(format!("Failed to canonicalize path: {path:?}"))
```

Comment:
Does this error show up often? It doesn't feel like it's worth having a special helper function just for this case.

Reply:
It does not, I reverted the code to simply use `path.canonicalize()?` ✅